pkg/ottl/grammar.go

```diff
@@ -270,6 +271,15 @@ type list struct {
 	Values []value `parser:"'[' (@@)* (',' @@)* ']'"`
 }
 
+type mapValue struct {
+	Values []mapItem `parser:"'{' (@@ ','?)* '}'"`
+}
+
+type mapItem struct {
+	Key   *string `parser:"@String ':'"`
+	Value *value  `parser:"@@"`
```

Comment:
Outside of `editor`, what's wrong with including those as values in a map literal?

Reply:
I don't think other languages allow `{example: 1+1}`. What would we do there? Evaluate it? Thats done as part of the Getter right now (hot path not parsing). If we don't evaluate it, it can't be put directly into a `pcommon.Map`.

Reply:
I'm nearly positive JavaScript also supports that syntax, but I double-checked and Go allows it:
```golang
	val := map[string]any{
		"test": 1 + 1,
	}

	fmt.Println(val) // Prints "map[test:2]"
```

I think these things are particularly useful when manipulating data in other paths, like in the test here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34168/files#diff-90b6e27c60a80c3c831576a03b3bd0a47839bfb606a9d2876d8ccf6bf1370c92R827.

Reply:
For some reason I wasn't thinking about these literals in the same context as your Go scenario but you're right, that is common.